cns/hnsclient/hnsclient_windows.go

```diff
@@ -301,7 +304,7 @@ func createHostNCApipaNetwork(
 		}
 
 		// Create loopback adapter needed for this HNS network
-		if interfaceExists, _ := networkcontainers.InterfaceExists(hostNCLoopbackAdapterName); !interfaceExists {
+		if interfaceExists, _ := networkcontainers.InterfaceExists(vEthernethostNCLoopbackAdapterName); !interfaceExists {
```

Comment:
@behzad-mir  @tamilmani1989  Say a loopback adapter has been created but hns network is still pending to be created. If in a case like if AzureContainerNetworking.exe gets invoked again, what would be the behavior? I am thinking it will fail but retries will succeed with this fix. Do we need to worry about such cases where calls can get interleaved?

Reply:
we have to test that scenario. @behzad-mir we can purposefully return error from cns after loopadapater creation before hns network creation and then retry should succeed

Reply:
The second ADD call only happens if HNSNetowrk and Loppback are created since the ADD calls are sequential.
The case that tHe loopback adapter exist but the HNS APipa network get deleted (which is the root cause of this issue) the fix was able to resolve the issue.